src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -753,10 +755,29 @@ ExecuteSingleModifyTask(CitusScanState *scanState, Task *task, bool multipleTask
 
 	char *queryString = task->queryString;
 	bool taskRequiresTwoPhaseCommit = (task->replicationModel == REPLICATION_MODEL_2PC);
+	bool failOnError = false;
 
 	ShardInterval *shardInterval = LoadShardInterval(task->anchorShardId);
 	Oid relationId = shardInterval->relationId;
 
+	if (multipleTasks)
+	{
+		taskRequiresTwoPhaseCommit |= (MultiShardCommitProtocol == COMMIT_PROTOCOL_2PC);
+	}
+
+	/* we don't allow failures to happen with CREATE INDEX CONCURRENTLY */
```

Comment:
I had to do that because we error out immediately with `COMMIT_PROTOCOL_BARE `, so wanted to stick to that (and pass the tests).

Reply:
Don't we always error out for DDL Commands? Should we check for DDL commands inside the loop?

Reply:
I think this is related to the comment you've done below. We use `COMMIT_PROTOCOL_2PC` to decide `failOnError`. But, `CREATE INDEX CONCURRENTLY` uses `COMMIT_PROTOCOL_BARE ` and to enable `failOnError` behavior, I added this hack.